tests/functional/test_performance.py

```diff
@@ -140,4 +185,40 @@ def test_basic_balance_on_whole_year() -> None:
     status = problem.solver.Solve()
 
     assert status == problem.solver.OPTIMAL
-    assert problem.solver.Objective().Value() == 3000 * horizon
+    assert problem.solver.Objective().Value() == 30 * 100 * horizon
+
+
+def test_basic_balance_on_whole_year_with_large_sum() -> None:
+    """
+    Balance on one node with one fixed demand and one generation with storage, on 8760 timestep.
+    """
+
+    scenarios = 1
+    horizon = 8760
+    time_block = TimeBlock(1, list(range(horizon)))
+
+    database = DataBase()
+    database.add_data("D", "demand", generate_const_data(100, horizon, scenarios))
+
+    database.add_data("G", "p_max", ConstantData(100))
```

Comment:
Why do we have 2 classes/methods for constant data ?

Reply:
I misnamed the function for lack of creativity.
While the second is a scalar value, the second is a (horizon, scenarios) matrix with the same value on every entry.

I accept name suggestions for the function :)